### PR TITLE
Malfunctioning AI Rework

### PR DESCRIPTION
**Pull request by Maldaris**
## About The Pull Request

Malfunctioning AI is in a bit of a weird state for low population. This PR will hopefully rework the traitor type to be less of a pain to deal with by removing it from low player rounds. In addition, objectives will favor obsession over murder/army/borg type objectives, where previously those were the norm.

To compensate for all this, a "Hack # APCs" objective is guaranteed to be added to the Malf AI objectives and the AI is given the ability to temporarily hide its presence in a single APC for a short period of time, and the expense of not being able to hack another APC while doing so.

## Why It's Good For The Game

The current Malf AI setup missed the re-balance pass of objectives that occurred with normal traitors, and it shows. This will hopefully bring it more inline with the rest of the traitor objective lineup/probabilities.

## Changelog
:cl:
add: Added Malfunctioning AI "Hack # APCs" objective
add: Added the ability for an AI to "mask" their presence in an APC for 30 to 60 seconds.
balance: Added minimum player count to spawn a Malfunctioning AI.
balance: Reworked objective layout similar to normal Traitors, with less likely murder objectives.
config: Added `MAX_MALF_APC_HACK_OBJ` and `TRAITOR_MALF_AI_MIN_POP` configuration values
/:cl:

### CODE REVIEW

File: code/modules/antagonists/traitor/datum_traitor.dm [hunk 1]
```diff
@@ -164,31 +165,34 @@
...
-			yandere_one.find_target()
-			var/datum/objective/maroon/yandere_two = new
-			yandere_two.owner = owner
-			yandere_two.target = yandere_one.target
-			yandere_two.update_explanation_text() // normally called in find_target()
-			add_objective(yandere_two)
-			.=2
+	// WaspStation Edit - Malf AI Rework
+	if(prob(20))
+		var/special_pick = rand(1,4)
```
Comment (MarkSuckerberg) [inline code/modules/antagonists/traitor/datum_traitor.dm:170]:
```suggestion
		var/special_pick = rand(1,3)
```

File: tgui/packages/tgui/interfaces/Apc.js [hunk 2]
```diff
@@ -167,6 +167,13 @@ export const Apc = props => {
                 color="bad"
                 onClick={() => act(malfStatus.action)} />
             )}
+            {!!data.malfStatus && data.malfStatus.action === 'occupy' && (
+              <Button
+                icon="eye-slash"
+                content="Mask Override Subroutines"
+                color="bad"
+                onClick={() => act("hack_hide")} />
+            )}
```
Comment (Superyodeler) [inline tgui/packages/tgui/interfaces/Apc.js:176]: The hack hiding function doesn't seem to be working. I think it's due to there being no entry in the malfMap const at the beginning of the file.

File: code/game/gamemodes/traitor/traitor.dm [hunk 3]
```diff
@@ -46,6 +52,9 @@
 					protected_jobs += "Chief Engineer"
 		restricted_jobs += protected_jobs
 
+	if(CONFIG_GET(number/traitor_malf_ai_min_pop) < total_ready_players)
+		restricted_jobs += "AI"
+
```
Comment (Superyodeler) [inline code/game/gamemodes/traitor/traitor.dm:57]: AI isn't appearing in restricted jobs even when I am the only player, not sure what's wrong with it.

**Comment by Superyodeler** [comment]
And this time I remembered to compile TGUI

File: code/game/gamemodes/traitor/traitor.dm (hunk 3 above)
Comment (Maldaris) [reply code/game/gamemodes/traitor/traitor.dm:57]: image
If you've not set the config value it won't restrict. Defaults to 1, ie off if not specified.

File: tgui/packages/tgui/interfaces/Apc.js (hunk 2 above)
Comment (Maldaris) [reply tgui/packages/tgui/interfaces/Apc.js:176]: `  {!!data.malfStatus && data.malfStatus.action === 'occupy' &&`

It's this bit failing. I wasn't 100% sure of what would need to be given as the condition so I might have to play with it a bit more.

**Comment by TheNeoGamer42** [comment]
okay so when is this gonna get fixed and merged fully

File: tgui/packages/tgui/interfaces/Apc.js (hunk 2 above)
Comment (Maldaris) [reply tgui/packages/tgui/interfaces/Apc.js]: fixed in 1666e42

File: code/game/gamemodes/traitor/traitor.dm (hunk 3 above)
Comment (Maldaris) [reply code/game/gamemodes/traitor/traitor.dm:57]: Please let me know if this issue persists.

File: config/game_options.txt [hunk 4]
```diff
@@ -224,6 +224,9 @@ NO_TRAITOR_HEAD CHIEF_MEDICAL_OFFICER
 NO_TRAITOR_HEAD RESEARCH_DIRECTOR
 NO_TRAITOR_HEAD CHIEF_ENGINEER
 
+## Minimum players joining the round required to create a malfunctioning AI
+TRAITOR_MALF_AI_MIN_POP 10
+
```
Comment (Superyodeler) [inline config/game_options.txt:229]: I've got this on my end, but it still doesn't seem to work for me. The restricted jobs line seems to work fine, so I think the problem is in the `if(CONFIG_GET(number/traitor_malf_ai_min_pop) < total_ready_players)` line. I've been messing with it for a while and I'm not sure what the problem is.

Comment (Maldaris) [reply config/game_options.txt:229]: fixed in d9040c960e

**Pull request merged**